hearpreprocess/util/audio.py

```diff
@@ -3,129 +3,168 @@
+        print(
+            "Skipping audio file for stats calculation. "
+            "Audio path: {file_path}"
+            f"Error: {e}"
+        )
+        audio_stats = {}
+    return audio_stats
 
 
-def audio_dir_stats_wav(
+def get_audio_dir_stats(
     in_dir: Union[str, Path], out_file: str, exts: Optional[List[str]] = None
 ):
-    """Produce summary by recursively searching a directory for wav files"""
+    """Produce summary by recursively searching a directory for audio files"""
     if exts is None:
-        exts = [".wav", ".mp3", ".ogg"]
+        exts = [".wav", ".mp3", ".ogg", ".webm"]
 
-    # Filter the files in the directory for the required extensions
+    # Get all the audio files
     audio_paths = list(
         filter(
             lambda audio_path: audio_path.suffix.lower()
             in map(str.lower, exts),  # type: ignore
             Path(in_dir).absolute().rglob("*"),
         )
     )
-    audio_dir_stats = list(
-        map(
-            audio_stats_wav,
-            tqdm(audio_paths),
-        )
-    )
 
+    # Count the number of successful and failed statistics extraction to be
+    # added the output stats file
+    success_counter: Dict[str, int] = defaultdict(int)
+    failure_counter: Dict[str, int] = defaultdict(int)
+
+    # Iterate and get the statistics for each audio
+    audio_dir_stats = []
+    for audio_path in tqdm(audio_paths):
+        audio_stats = get_audio_stats(audio_path)
+        if audio_stats:
+            audio_dir_stats.append(audio_stats)
+            success_counter[audio_path.suffix] += 1
+        else:
+            # update the failed counter if the extraction was not
+            # succesful
+            failure_counter[audio_path.suffix] += 1
+
+    assert audio_dir_stats, "Stats was not calculated for any audio file. Please Check"
+    " the formats of the audio file"
     durations = [stats["duration"] for stats in audio_dir_stats]
     unique_sample_rates = dict(
         Counter([stats["sample_rate"] for stats in audio_dir_stats])
     )
+    mono_audio_count = sum(stats["mono"] for stats in audio_dir_stats)
 
-    stats = {
-        "audio_count": len(durations),
+    summary_stats = {
+        # Count of no of success and failure for audio summary extraction for each
+        # extension type
+        "audio_summary_from": {
+            "successfully_extracted": success_counter,
+            "failed_to_extract": failure_counter,
+        },
         "audio_samplerate_count": unique_sample_rates,
+        "mono_audio_count": mono_audio_count,
         "audio_mean_dur(sec)": np.mean(durations),
         "audio_median_dur(sec)": np.median(durations),
-    }
-    stats.update(
-        {
+        # Percentile duration of the audio
+        **{
             f"{str(p)}th percentile dur(sec)": np.percentile(durations, p)
             for p in [10, 25, 75, 90]
-        }
-    )
-    json.dump(stats, open(out_file, "w"), indent=True)
+        },
+    }
```

Comment:
1) Can you put the label counts in this file too?

2) Can you combine all the files at the end with one final job so there is like one JSON file that contains like all the information about a task preprocessing, so you can quickly look at one file and debug if it's correct?

Reply:
For 2 above, we'll do it separately using this https://github.com/neuralaudio/hear-preprocess/issues/24